lib/modelManagers/articlesComment.js

```diff
@@ -0,0 +1,92 @@
+// const user = require('../../database/models').User;
+// const articles = require('../../database/models');
+
+const Sequelize = require('sequelize');
+const { articlesComment } = require('../../database/models');
+// const Sequelize = require('sequelize');
+
+const { Op } = Sequelize;
+
+/**
+ * @description articlesComment Model Manager
+ * @class articlesComment
+ */
+class articleComment {
+  /**
+   *
+   * @description This method create articles comment
+   * @static
+   * @param {*} comment
+   * @param {*} userId
+   * @param {*} articleId
+   * @param {*} mainCommentId
+   * @memberof articlesComment
+   * @returns {*} createdComment
+   */
+  static async createComment(comment, userId, articleId, mainCommentId) {
+    const createdComment = await articlesComment.create({
+      comment,
+      userId,
+      articleId,
+      mainCommentId
+    });
+    return createdComment;
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} articleId
+   * @param {*} offset
+   * @returns {*} comments
+   * @memberof articleComment
+   */
+  static async findAllComment(articleId, offset = 1) {
+    const options = {
+      where: {
+        mainCommentId: {
+          [Op.eq]: null
+        },
+        articleId
+      },
+      include: {
+        model: articlesComment,
+        as: 'threads'
+      },
+      limit: 5 * Number(offset)
+    };
+    const comments = await articlesComment.findAll(options);
+    return comments;
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} articleId
+   * @param {*} commentId
+   * @param {*} offset
+   * @returns {*} *
+   * @memberof articleComment
+   */
+  static async findCommentThreads(articleId, commentId, offset = 1) {
+    const options = {
+      where: {
+        id: {
+          [Op.eq]: commentId
+        },
+        articleId
+      },
```

Comment:
Can we have only main comment id returned here?

Reply:
yes the commentId is the mainCommentId in the params that must match with the Id of the comment to be searched for